mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -108,7 +108,7 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
     }
     if (getPredictionCol != "") {
       val predUDF = if (getRawPredictionCol != "") {
-        callUDF(raw2prediction _, DoubleType, col(getRawPredictionCol))
+        udf[Double, Vector](raw2prediction).apply(col(getRawPredictionCol))
```

Comment:
The type info is not necessary. Try

~~~scala
udf(raw2prediction _).apply(col(getRawPredictionCol))
~~~

The `callUDF` on line 106 needs some explicit type info since `FeaturesType` doesn't have type tag. We can write this:

~~~scala
      val predictRawUDF = udf { (features: Any) =>
        predictRaw(features.asInstanceOf[FeaturesType])
      }
      outputData = outputData.withColumn(getRawPredictionCol,
        predictRawUDF(col(getFeaturesCol)))
~~~